subsys/net/lib/mqtt_socket/mqtt_decoder.c

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file mqtt_decoder.c
+ *
+ * @brief Decoder functions needs for decoding packets received from the broker.
+ */
+
+#define LOG_MODULE_NAME net_mqtt_dec
+#define NET_LOG_LEVEL CONFIG_MQTT_LOG_LEVEL
+
+#include "mqtt_internal.h"
+#include "mqtt_os.h"
+
+#define MQTT_LENGTH_VALUE_MASK       0x7F
+#define MQTT_LENGTH_CONTINUATION_BIT 0x80
+#define MQTT_LENGTH_MULTIPLIER       0x80
```

Comment:
I'm not sure what exactly do you mean here, `#define MQTT_LENGTH_CONTINUATION_BIT (1 << 7)`?

Reply:
Nope, I meant `#define MQTT_LENGTH_SHIFT       7` and then use `<<` instead of `*`, like in old good days of C (the code of this PR gives "Java" feeling in many places).

Of course, that's relatively minor comparing to other stuff, ~ personal opinion actually.